airbyte-integrations/connectors/source-netsuite/source_netsuite/source.py

```diff
@@ -20,6 +20,11 @@ class SourceNetsuite(AbstractSource):
 
     logger: logging.Logger = logging.getLogger("airbyte")
 
+    def _get_default_date_format(self, config: Mapping[str, Any]) -> str:
```

Comment:
issue: You probably want to validate what you get from the config. Also, return type should be `Optional[str]` as you can return `None`.

Reply:
Normally, if it's easy, I would use property's pattern field in `spec.yaml` file. However, there is not an easy regexp to validate a given date format. So, we need to validate it using source code of the connector. The validation of date format is done in `streams.py L196`. 
Another reason for not putting a validation for this is the general behavior of the connector for parameters. For instance, you can also validate `object_types` using Netsuite API but they do not do that. 
Agree with you about parameter validation but it seems a bit over-killing for me. WDYT?